### PR TITLE
adds shell.nix

### PR DESCRIPTION
**Pull request by legendofmiracles**
@buffet i was asked to ask you to check if it's alright

It also requires nixpkgs-unstable :/

### CODE REVIEW

**Review by buffet — COMMENTED** [review]
Do you need rustc when you have cargo in there?

openssl should be buildInputs (non-native), not sure sqlx, I assume it gets used in the build process, so should be fine.

**Comment by legendofmiracles** [comment]
Yes you do need rustc, in case you have another version installed outside the nix-shell. As far as I am aware it also doesn't really add more dependencies, as cargo and rustc already point to the same version. 

And sqlx-cli isn't necessarily needed for building, only for changing the DB. So I'm unsure where to put that.

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{ pkgs ? import <nixpkgs> {}  }:
+pkgs.mkShell {
+          nativeBuildInputs = with pkgs; [ rustc cargo gcc openssl pkg-config sqlx-cli ];
```
Comment (fortuneteller2k) [inline shell.nix:3]:
```suggestion
          nativeBuildInputs = with pkgs; [ rustc cargo gcc openssl pkg-config ];
          
          buildInputs = with pkgs; [ sqlx-cli ];
```

**Review by fortuneteller2k — CHANGES_REQUESTED** [review]
runtime dependencies should be in `buildInputs`

**Comment by legendofmiracles** [comment]
Ok cool! Thank you so much!

**Review by fortuneteller2k — APPROVED** [review]

**Comment by elkowar** [comment]
Epic, thanks! Also thanks to @buffet for the review

**Comment by legendofmiracles** [comment]
And @fortuneteller2k.

**Pull request merged**